tracer/test/Datadog.Trace.Tests/SpanContextPropagatorTests_ExtractBaggageHeaderTags.cs

```diff
@@ -0,0 +1,142 @@
+// <copyright file="SpanContextPropagatorTests_ExtractBaggageHeaderTags.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.TestHelpers.TestTracer;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests
+{
+    [Collection(nameof(WebRequestCollection))]
+    public class SpanContextPropagatorTests_ExtractBaggageHeaderTags
+    {
+        public enum HeaderCollectionType
+        {
+            /// <summary>
+            /// NameValueCollection
+            /// </summary>
+            NameValueHeadersCollection,
+
+            /// <summary>
+            /// WebHeadersCollection
+            /// </summary>
+            WebHeadersCollection,
+        }
+
+        public static TheoryData<HeaderCollectionType> GetHeaderCollections()
+            => new() { HeaderCollectionType.NameValueHeadersCollection, HeaderCollectionType.WebHeadersCollection, };
+
+        internal static IHeadersCollection GetHeadersCollection(HeaderCollectionType type)
+            => type switch
+            {
+                HeaderCollectionType.WebHeadersCollection => WebRequest.CreateHttp("http://localhost").Headers.Wrap(),
+                HeaderCollectionType.NameValueHeadersCollection => new NameValueCollection().Wrap(),
+                _ => throw new Exception("Unknown header collection type " + type),
+            };
+
+        [Fact]
+        internal async Task AddBaggageTags()
+        {
+            // Add Baggage items using a key that's in the default configuration
+            var baggageItems = new Dictionary<string, string>();
+            baggageItems.Add("user.id", "doggo");
+            var baggage = new Baggage(baggageItems);
+
+            var expectedResult = "doggo";
```

Comment:
super small nit: I think you can remove this and just set the expectedResult below to `"doggo"`